### ISSUE TITLE
Resize function improvement of Variable Explorer

### ISSUE THREAD

**Issue by mIgLLL** [spyder-ide/ux-improvements#65]
### Issue Report Checklist

* [ ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [ ] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [ ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
I find that the 'Resize' function is not so useful since what I wanted when I use the 'Resize' is that I want to take a rough look on the dataset. So, two kind of situation maybe suitable: 1, spyder help me 'Resize' all the variable name length for me to look at all of the variable names fully; 2, spyder help me 'Resize' all the data length for me to look at all of the data themselves fully.

I would show the difference between the original situation and the two suitable situations I proposed.

Original data surface:
![image](https://user-images.githubusercontent.com/31037879/138548211-602d764b-aa05-453d-ad57-4d642df3794f.png)
This dataset can not show the second situable situation so I add a "test" variable with just a text "aaaaaaaaaaaaaaaa"

Original after 'Resize' surface:
![image](https://user-images.githubusercontent.com/31037879/138548301-08aaee34-d643-4ae2-9e86-e0cf31d1215e.png)
I guess this 'Resize' buttom want to show the suitable situation 1, helping users to take a look at the variable names easily. But just don't work.

Suitable situaion 1: (intend for a full look of variable names)
![image](https://user-images.githubusercontent.com/31037879/138548366-34b53dec-354b-4fd0-a242-9ff3b2628a69.png)
I don't adjust the variable names precisely but I think this is not a confusing description.

Suitable situation 2: (intend for a full look of data but also the variable names)
![image](https://user-images.githubusercontent.com/31037879/138548456-67b23c28-4217-4141-b0e7-f0ea846076bf.png)
I think the second situation is quite useful unless there is an unexpected long variable such as website "http://github.com/spyder/spyder/spyder/spyder/spyder/spyder/spyder/spyder/spyder/spyder/spyder/spyder/spyder", this may make the variable explorer too bloated. But I think this is still a good function, let users choose which to use is a good way.

In conclusion, I think the 'resize' function of variable explorer should be improved into two kinds of adjustments like I described above. 

### What steps reproduce the problem?

1. 
2. 
3. 

### Paste Traceback/Error Below (if applicable)

```python-traceback

PASTE TRACEBACK HERE

```

## Versions

* Spyder version: 5.1.5
* Python version: 3.8
* Qt version: 
* PyQt version: 
* Operating System name/version: Win10 19042
Resolution: 3840x2160
### Dependencies

```
PASTE DEPENDENCIES HERE
```
